src/ApplicationWindow.vala

```diff
@@ -18,21 +18,80 @@
 [GtkTemplate (ui = "/com/github/nahuelwexd/Replay/ApplicationWindow.ui")]
 public class Rpy.ApplicationWindow : Hdy.ApplicationWindow
 {
-	[GtkChild] private Rpy.HeaderBar _header_bar;
+	[GtkChild] private unowned Hdy.Leaflet _leaflet;
 
+	private Rpy.NavigationService? _navigation_service;
 
-	public ApplicationWindow (Rpy.Application app)
+
+	public Rpy.NavigationService? navigation_service
+	{
+		get { return this._navigation_service; }
+		construct
+		{
+			if (value == null)
+			{
+				return;
+			}
+
+			this._navigation_service = value;
+
+			((!) this._navigation_service).go_back.connect (this.go_back);
```

Comment:
@nahuelwexd what does the ((!) here do?

Reply:
I'm using the `--enable-experimental-non-null` flag, but due to Vala lacking [this](https://gitlab.gnome.org/GNOME/vala/-/issues/1058) I need to cast the nullable type to a non-nullable type manually using `(!)`